src/python/pants/fs/archive.py

```diff
@@ -22,8 +22,28 @@
 class Archiver(AbstractClass):
 
   @classmethod
-  def extract(cls, path, outdir):
-    """Extracts an archive's contents to the specified outdir."""
+  def extract(cls, path, outdir, filter_func=None, safe=False):
+    """Extracts an archive's contents to the specified outdir with an optional filter.
+
+    :API: public
+
+    :param string path: path to the zipfile to extract from
+    :param string outdir: directory to extract files into
+    :param function filter_func: optional filter with the filename as the parameter.  Returns True
+      if the file should be extracted.  Note that filter_func is ignored for non-zip archives.
+    :param bool safe: True if use safe extraction.  Safe extraction ignores errors if destination
```

Comment:
Maybe `concurrency_safe` or `safe_concurrent`, to indicate what the parameter is guarding against?

Reply:
But actually, thinking about it more, it's not clear that this needs to be a parameter. The rename should be nearly zero cost.